segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java

```diff
@@ -181,55 +181,40 @@ default int maximumValueLength() {
     CompletableFuture<List<TableEntry>> get(String segmentName, List<ArrayView> keys, Duration timeout);
 
     /**
-     * Creates a new Iterator over all the {@link TableKey}s in the given Table Segment.
+     * Creates a new Iterator over all the {@link TableKey} instances the given Table Segment.
      *
-     * @param segmentName The name of the Table Segment to iterate over.
-     * @param state       An {@link IteratorState} that can be used to resume a previously interrupted iteration. This
-     *                    can be obtained by invoking {@link IteratorItem#getState()}.
-     * @param timeout     Timeout for the operation.
+     * @param segmentName     The name of the Table Segment to iterate over.
+     * @param serializedState (Optional) A byte array representing the serialized form of the State. This can be obtained
+     *                        from {@link IteratorItem#getState()}. If provided, the iteration will resume from where it
+     *                        left off, otherwise it will start from the beginning.
+     * @param fetchTimeout    Timeout for each invocation to {@link AsyncIterator#getNext()}.
      * @return A CompletableFuture that, when completed, will return an {@link AsyncIterator} that can be used to iterate
-     * over all the {@link TableKey} instances in the given Table Segment. If the operation failed, the Future will be
-     * failed with the causing exception. Notable exceptions:
+     * over all the {@link TableKey} instances in the Table. If the operation failed, the Future will be failed with the
+     * causing exception. Notable exceptions:
      * <ul>
      * <li>{@link StreamSegmentNotExistsException} If the Table Segment does not exist.
      * <li>{@link BadSegmentTypeException} If segmentName refers to a non-Table Segment.
+     * <li>{@link java.io.IOException} If serializedState is non-null but it cannot be deserialized.
```

Comment:
1. It throws IOException because it uses `VersionedSerializer` to deserialize the given byte array. That can be made of `SerializationException` (bad format), `EOFException` (insufficient data), or some other unexpected `IOException`.
2. This shouldn't be `DataCorruptionException` because it does not indicate internal data corruption. This deals with the argument `serializedState` which is passed in by the user.

I do see your point. This is an error referring to an argument. I changed it to an `IllegalDataFormatException` (which extends `IllegalArgumentException`)) and pass in the `IOException` as root cause. That is more in line with how a user might expect this to work.